server/apps/publish/formatters/field_mappers/selectorcode_mapper.py

```diff
@@ -82,92 +140,146 @@ def map(self, article, category, **kwargs):
             handler = handlers.get(desk['name'].upper(), lambda *args: None)
             handler(article, subscriber['name'], formatted_item, category)
 
+    def _get_geo_abbreviation(self, name):
+        """
+        passed the name that is kept in the article uses it to find the value from the vocabulary
+        :param name:
+        :return: the abbreviatoin for the geographical restriction
+        """
+        geo_resource = superdesk.get_resource_service('vocabularies').find_one(req=None,
+                                                                               _id='geographical_restrictions')
+        item = [i for i in geo_resource.get('items') if i['name'] == name][0]
```

Comment:
Depending on the number of items, using an iterator can be quite a bit faster than constructing a whole list.
```py
item = next(i for i in geo_resource.get('items') if i['name'] == name, None)
```